pkg/registry/flowcontrol/rest/storage_flowcontrol.go

```diff
@@ -0,0 +1,148 @@
+
+	if apiResourceConfigSource.VersionEnabled(flowcontrolv1alpha1.SchemeGroupVersion) {
+		apiGroupInfo.VersionedResourcesStorageMap[flowcontrolv1alpha1.SchemeGroupVersion.Version] = p.v1alpha1Storage(apiResourceConfigSource, restOptionsGetter)
+	}
+	return apiGroupInfo, true
+}
+
+func (p RESTStorageProvider) v1alpha1Storage(apiResourceConfigSource serverstorage.APIResourceConfigSource, restOptionsGetter generic.RESTOptionsGetter) map[string]rest.Storage {
+	storage := map[string]rest.Storage{}
+
+	// flow-schema
+	flowSchemaStorage, flowSchemaStatusStorage := flowschemastore.NewREST(restOptionsGetter)
+	storage["flowschemas"] = flowSchemaStorage
+	storage["flowschemas/status"] = flowSchemaStatusStorage
+
+	// priority-level-configuration
+	priorityLevelConfigurationStorage, priorityLevelConfigurationStatusStorage := prioritylevelconfigurationstore.NewREST(restOptionsGetter)
+	storage["prioritylevelconfigurations"] = priorityLevelConfigurationStorage
+	storage["prioritylevelconfigurations/status"] = priorityLevelConfigurationStatusStorage
+
+	return storage
+}
+
+func (p RESTStorageProvider) PostStartHook() (string, genericapiserver.PostStartHookFunc, error) {
+	systemPreset := SystemPresetData{
+		FlowSchemas:                 bootstrap.SystemFlowSchemas(),
+		PriorityLevelConfigurations: bootstrap.SystemPriorityLevelConfigurations(),
+	}
+	// TODO: default flow-schemas and priority levels
+	return PostStartHookName, systemPreset.EnsureSystemPresetConfiguration(), nil
+}
+
+type SystemPresetData struct {
+	FlowSchemas                 []*flowcontrolv1alpha1.FlowSchema
+	PriorityLevelConfigurations []*flowcontrolv1alpha1.PriorityLevelConfiguration
+}
+
+func (d SystemPresetData) EnsureSystemPresetConfiguration() genericapiserver.PostStartHookFunc {
+	return func(hookContext genericapiserver.PostStartHookContext) error {
+		flowcontrolClientSet := flowcontrolclient.NewForConfigOrDie(hookContext.LoopbackClientConfig)
+		// Adding system priority classes is important. If they fail to add, many critical system
+		// components may fail and cluster may break.
+		err := wait.Poll(1*time.Second, 30*time.Second, func() (done bool, err error) {
+			if err != nil {
+				utilruntime.HandleError(fmt.Errorf("unable to initialize client: %v", err))
+				return false, nil
+			}
+
+			for _, flowSchema := range d.FlowSchemas {
+				_, err := flowcontrolClientSet.FlowSchemas().Get(flowSchema.Name, metav1.GetOptions{})
+				if err != nil {
+					if apierrors.IsNotFound(err) {
+						_, err := flowcontrolClientSet.FlowSchemas().Create(flowSchema)
+						if err != nil && !apierrors.IsAlreadyExists(err) {
+							return false, err
+						} else if err == nil {
+							klog.V(2).Infof("created system preset FlowSchema %s", flowSchema.Name)
+						} else {
+							klog.V(2).Infof("system preset FlowSchema %s already exists, skipping creating", flowSchema.Name)
```

Comment:
There is another "already exists" condition.  It seems odd that one gets a log message while the other does not.  Also, `V(2)` seems a bit prominent for a message saying nothing happened.

Reply:
i adjusted verbosity to 6